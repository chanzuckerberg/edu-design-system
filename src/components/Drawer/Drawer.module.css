@import '../../design-tokens/mixins.css';

/*------------------------------------*\
    # DRAWER
\*------------------------------------*/

/**
 * 1) Box that overlays the application's content, prompting a user to perform an action.
 */
.drawer__container {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: flex-end;
  visibility: hidden;
  width: 28rem;
  height: 100vh;
  position: fixed;
  top: 0;
  right: 0;
  z-index: var(--eds-z-index-bottom);
  background-color: transparent;
}

.drawer__container:focus-visible {
  @mixin focus;
}

/**
* Active drawer container
*/
.drawer__container--is-active {
  visibility: visible;
  z-index: var(--eds-z-index-400);
}

/**
 * Drawer window
 * 1) Actual box with content within
 */
.drawer {
  display: flex;
  flex-direction: column;
  height: 100vh;
  width: 100%;
  transform: translateX(100%);
  background: var(--eds-theme-color-background-neutral-default);
  border-left-width: var(--eds-theme-border-width);
  border-left-style: solid;
  border-left-color: var(--eds-theme-color-border-neutral-subtle);
  box-shadow: var(--eds-box-shadow-md);
  transition: transform var(--eds-anim-move-quick) var(--eds-anim-ease);

  @media screen and (prefers-reduced-motion) {
    transition: none;
  }
}
.drawer:focus-visible {
  @mixin focus;
}

/**
* Drawer window inside active drawer container
* 1) Transform added for animation
*/
.drawer__container--is-active .drawer {
  transform: translateX(0); /* 1 */
}

/**
 * Modal header
 */
.drawer__header {
  flex-shrink: 0;
  height: var(--eds-size-9);
  border-bottom: var(--eds-theme-border-width) solid
    var(--eds-theme-color-border-neutral-subtle);

  display: flex;
  align-items: center;
  justify-content: space-between;
}

/**
 * Drawer body
 */
.drawer__body {
  flex: 1;
}
