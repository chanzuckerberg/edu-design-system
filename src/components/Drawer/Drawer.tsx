import clsx from 'clsx';
import React, {
  useCallback,
  useState,
  useEffect,
  useRef,
  type ReactNode,
  type KeyboardEvent,
} from 'react';
import { oneByType } from 'react-children-by-type';
import FocusLock from 'react-focus-lock';
import { Portal } from 'react-portal';
import styles from './Drawer.module.css';
import { ESCAPE_KEYCODE } from '../../util/keycodes';
import DrawerBody from '../DrawerBody';
import DrawerFooter from '../DrawerFooter';
import DrawerHeader from '../DrawerHeader';

export type Props = {
  /**
   * HTML id of the helper text used to describe the component after aria-labelledby
   */
  'aria-describedby'?: string;
  /**
   * HTML id of the helper text used to label the drawer component
   */
  'aria-labelledby'?: string;
  /**
   * Child node(s) that can be nested inside component. `DrawerHeader`, `DrawerBody`, and `ModelFooter` are the only permissible children of the Drawer
   */
  children?: ReactNode;
  /**
   * CSS class names that can be appended to the Drawer container component that the Drawer components reside in.
   */
  className?: string;
  /**
   * Toggles the ability to dismiss the Drawer window
   */
  dismissible?: boolean;
  /**
   * CSS class names that can be appended to the Drawer wrapper component that the Drawer window translates in and out of.
   */
  drawerContainerClassName?: string;
  /**
   * Sets the drawer to open or close by default
   */
  isActive?: boolean;
  /**
   * Handler to be called when the drawer is being closed (by ESCAPE / clicking X / clicking outside)
   */
  onClose?: (e?: any) => void;
};

/**
 * BETA: This component is still a work in progress and is subject to change.
 *
 * `import {Drawer} from "@chanzuckerberg/eds";`
 *
 * A window component that slides from and out of the right side of the screen.
 */
export const Drawer = ({
  'aria-describedby': ariaDescribedBy,
  'aria-labelledby': ariaLabelledBy,
  drawerContainerClassName,
  isActive,
  children,
  dismissible,
  onClose,
  className,
  ...other
}: Props) => {
  /**
   * Initialize states, constants, and refs
   */
  const [activeFocus, setActiveFocus] = useState(isActive);
  const windowRef = useRef<HTMLElement | null>(null);
  const isMountedRef = useRef(true);
  const onCloseRef = useRef(onClose);

  /**
   * Update activeFocus to be consistent with isActive.
   */
  useEffect(() => {
    if (isActive) {
      activateFocusTrap();
    } else {
      deactivateFocusTrap();
    }
  }, [isActive]);

  /**
   * Set a flag when this item has mounted
   * Otherwise, portals will render which are not handled well
   * between React and Next.js
   */
  useEffect(() => {
    isMountedRef.current = true;
    return () => {
      isMountedRef.current = false;
    };
  }, []);

  useEffect(() => {
    onCloseRef.current = onClose;
  }, [onClose]);

  /**
   * Open the Drawer and give it focus.
   */
  function activateFocusTrap() {
    /**
     * This accommodates drawer animation so that it auto receives focus
     */
    setTimeout(() => {
      if (isMountedRef.current) {
        setActiveFocus(true);
      }
    }, 300);
  }

  /**
   * Close the drawer and remove focus.
   */
  function deactivateFocusTrap() {
    setActiveFocus(false);
  }

  /**
   * Close the drawer and run the onClose prop (pass in function) if it exists.
   */
  const handleOnClose = useCallback(() => {
    deactivateFocusTrap();

    if (onCloseRef.current) {
      onCloseRef.current();
    }
  }, []);

  /**
   * Handle "click outside"
   */
  useEffect(() => {
    function handleOnClickOutside(e: MouseEvent) {
      if (
        isActive &&
        dismissible &&
        windowRef.current &&
        !windowRef.current.contains(e.target as HTMLElement)
      ) {
        /**
         * onClick of the area around the drawer window, close the drawer
         */
        handleOnClose();
      }
    }
    document.addEventListener('click', handleOnClickOutside);
    return () => {
      document.removeEventListener('click', handleOnClickOutside);
    };
  }, [isActive, dismissible, handleOnClose, windowRef]);

  /**
   * If escape button is struck, close the drawer
   */
  function handleOnKeyDown(e: KeyboardEvent<HTMLElement>) {
    if (e.key === ESCAPE_KEYCODE) {
      handleOnClose();
    }
  }

  const drawerHeader = oneByType(children, DrawerHeader);
  const header = React.Children.map(drawerHeader, (child) => {
    return React.cloneElement(child, {
      onClick: () => handleOnClose(),
      dismissible: dismissible,
    });
  });
  const body = oneByType(children, DrawerBody);
  const footer = oneByType(children, DrawerFooter);

  const containerClassName = clsx(
    styles['drawer__container'],
    isActive && styles['drawer__container--is-active'],
    drawerContainerClassName,
  );

  const componentClassName = clsx(styles['drawer'], className);

  if (!isMountedRef.current) return null;

  return (
    <Portal>
      <FocusLock disabled={!activeFocus} returnFocus>
        {/**
         * the drawer will probably have buttons inside,
         * We're intentionally not adding role=button for now,
         * may so we're comfortable suppressing this rule
         */}
        {/* eslint-disable-next-line jsx-a11y/no-static-element-interactions */}
        <div
          aria-hidden={!isActive}
          className={containerClassName}
          onKeyDown={handleOnKeyDown}
          {...other}
        >
          <article
            aria-describedby={ariaDescribedBy}
            aria-labelledby={ariaLabelledBy}
            aria-modal={isActive}
            className={componentClassName}
            ref={windowRef}
            role="dialog"
            tabIndex={0} // eslint-disable-line jsx-a11y/no-noninteractive-tabindex
          >
            {header}
            {body}
            {footer}
          </article>
        </div>
      </FocusLock>
    </Portal>
  );
};

Drawer.Body = DrawerBody;
Drawer.Footer = DrawerFooter;
Drawer.Header = DrawerHeader;
