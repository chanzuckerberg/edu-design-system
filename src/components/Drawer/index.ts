export { Drawer as default } from './Drawer';
