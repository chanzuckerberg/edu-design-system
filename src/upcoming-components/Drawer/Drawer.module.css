@import '../../design-tokens/mixins.css';

/*------------------------------------*\
    # DRAWER
\*------------------------------------*/

/**
 * 1) Box that overlays the application's content, prompting a user to perform an action
 * 2) This transform rule stops the background overlay (present when modal is open)
 *    from flickering when a button is hovered within the modal
 */
.drawer {
  display: flex;
  justify-content: flex-end;
  visibility: hidden;
  width: 100%;
  height: 100vh;
  position: fixed;
  top: 0;
  right: 0;
  z-index: var(--eds-z-index-bottom);
  opacity: 0;
  overflow: auto;
  background-color: transparent;

  /**
  * Active modal
  */
  &.eds-is-active {
    opacity: 1;
    visibility: visible;
    z-index: var(--eds-z-index-400);
  }

  .drawer--show-backdrop & {
    background-color: rgba(0, 0, 0, 0.4);
  }

  &:focus-visible {
    @mixin focus;
  }
}

/**
 * Modal window
 * 1) Actual box with content within
 */
.drawer__window {
  display: flex;
  flex-direction: column;
  position: fixed;
  top: 0;
  left: right;
  height: 100vh;
  width: 28rem;
  max-width: 100%;
  transform: translateX(28rem);
  background: var(--eds-theme-color-background-neutral-default);
  border-left-width: var(--eds-theme-border-width);
  border-left-style: solid;
  border-left-color: var(--eds-theme-color-border-neutral-subtle);
  box-shadow: var(--eds-box-shadow-md);
  transition: transform var(--eds-anim-move-quick) var(--eds-anim-ease);

  &:focus-visible {
    @mixin focus;
  }

  @media screen and (prefers-reduced-motion) {
    transition: none;
  }

  /**
  * Modal window inside active modal
  * 1) Transform added for animation
  */
  .drawer.eds-is-active & {
    transform: translateX(0); /* 1 */
  }
}

/**
* Modal close button
*/
.drawer__close-button {
  margin-left: auto;

  @media all and (min-width: $eds-bp-sm) {
    display: block;
    position: absolute;
    top: var(--eds-size-2);
    right: var(--eds-size-2);
  }
}

/**
 * Modal header
 */
.drawer__header {
  display: flex;
  padding: var(--eds-size-2) var(--eds-size-3);
}

/**
 * Modal body
 */
.drawer__body {
  padding: var(--eds-size-3);
  flex: 1;
}

/**
 * Modal footer
 */
.drawer__footer {
  padding: 0 var(--eds-size-3) var(--eds-size-3);
}
