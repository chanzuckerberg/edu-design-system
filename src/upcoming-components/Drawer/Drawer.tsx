import clsx from 'clsx';
import React, {
  useState,
  useEffect,
  useRef,
  ReactNode,
  KeyboardEvent,
  MouseEvent,
} from 'react';
import { oneByType } from 'react-children-by-type';
import FocusLock from 'react-focus-lock';
import { Portal } from 'react-portal';
import styles from './Drawer.module.css';
import { ESCAPE_KEYCODE } from '../../util/keycodes';
import DrawerBody from '../DrawerBody';
import DrawerFooter from '../DrawerFooter';
import DrawerHeader from '../DrawerHeader';

export interface Props {
  /**
   * HTML id of the helper text used to describe the component after ariaLabelledBy
   */
  ariaDescribedBy?: string;
  /**
   * HTML id of the helper text used to label the drawer component
   */
  ariaLabelledBy?: string;
  /**
   * Child node(s) that can be nested inside component. `DrawerHeader`, `DrawerBody`, and `ModelFooter` are the only permissible children of the Drawer
   */
  children?: ReactNode;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
  /**
   * Button text for the drawer close button. This is visibly hidden but announced to screen reader users.
   */
  closeButtonText?: string;
  /**
   * Toggles the ability to dismiss the Drawer window
   */
  dismissible?: boolean;
  /**
   * Sets the drawer to open or close by default
   */
  isActive?: boolean;
  /**
   * Toggles the shadowed backdrop of the drawer on or off for flexibility
   */
  showBackdrop?: boolean;
  /**
   * Handler to be called when the drawer is being closed (by ESCAPE / clicking X / clicking outside)
   */
  onClose?: (e?: any) => void;
}

/**
 * BETA: This component is still a work in progress and is subject to change.
 *
 * ```ts
 * import {Drawer} from "@chanzuckerberg/eds";
 * ```
 *
 * TODO: update this comment with a description of the component.
 */
export const Drawer = ({
  ariaDescribedBy,
  ariaLabelledBy,
  className,
  isActive,
  children,
  dismissible,
  onClose,
  closeButtonText,
  showBackdrop = false,
  ...other
}: Props) => {
  /**
   * Initialize states, constants, and refs
   */
  const [isMounted, setIsMounted] = useState(false);
  const BODY_DISABLED_CLASS = `eds-body-is-disabled`;
  const [activeFocus, setActiveFocus] = useState(isActive || false);
  const windowRef = useRef<HTMLElement | null>(null);
  const ref = useRef<HTMLDivElement | null>(null);

  /**
   * Get previous prop
   * 1) This is used to compare the previous prop to the current prop
   *
   * TODO: improve `any` type
   */
  function usePrevious(isActive: any) {
    useEffect(() => {
      ref.current = isActive;
    });
    return ref.current;
  }

  /**
   * Use effect
   * 1) Set prevIsActive to previous isActive prop
   * 2) If prevIsActive is defined and previous isActive prop is not equal
   * to current isActive prop, toggle state
   */
  const prevIsActive = usePrevious(isActive); /* 1 */
  useEffect(() => {
    if (prevIsActive !== undefined || null) {
      if (isActive) {
        activateDOM();
      } else {
        deactivateDOM();
      }
    }
  });

  /**
   * Set a flag when this item has mounted
   * Otherwise, portals will render which are not handled well
   * between React and Next.js
   */
  useEffect(() => {
    setIsMounted(true);
  }, [setIsMounted]);

  /**
   * Activate DOM
   * 1) Open the Drawer
   * 2) Set activeFocus state to true
   * 3) This accommodates drawer animation so that it auto receives focus
   */
  function activateDOM() {
    document.body.classList.add(BODY_DISABLED_CLASS);

    /* 3 */
    setTimeout(() => {
      setActiveFocus(true); /* 2 */
    }, 300);
  }

  /**
   * Deactivate DOM
   * 1) Close the drawer
   * 2) Set activeFocus state to false
   */
  function deactivateDOM() {
    document.body.classList.remove(BODY_DISABLED_CLASS);
    setActiveFocus(false); /* 2 */
  }

  /**
   * Handle onClose
   * 1) Close the drawer
   * 2) Run the onClose prop (pass in function) if it exists
   */
  function handleOnClose() {
    deactivateDOM(); /* 1 */

    if (onClose) {
      onClose(); /* 2 */
    }
  }

  /**
   * Handle "click outside"
   * 1) onClick of the area around the drawer window, close the drawer
   */
  function handleOnClickOutside(e: MouseEvent<HTMLElement>) {
    if (
      isActive &&
      dismissible &&
      windowRef.current &&
      !windowRef.current.contains(e.target as HTMLElement)
    ) {
      handleOnClose(); /* 1 */
    }
  }

  /**
   * Handle onKeyDown
   * 1) If escape button is struck, close the drawer
   */
  function handleOnKeyDown(e: KeyboardEvent<HTMLElement>) {
    if (e.key === ESCAPE_KEYCODE) {
      handleOnClose(); /* 1 */
    }
  }

  const drawerHeader = oneByType(children, DrawerHeader);
  const header = React.Children.map(drawerHeader, (child) => {
    return React.cloneElement(child, {
      onClick: () => handleOnClose(),
      dismissible: dismissible,
    });
  });
  const body = oneByType(children, DrawerBody);
  const footer = oneByType(children, DrawerFooter);

  const componentClassName = clsx(
    styles['drawer'],
    className,
    showBackdrop && styles['drawer--show-backdrop'],
    isActive && styles['eds-is-active'],
  );

  if (!isMounted) return null;

  return (
    <Portal>
      <FocusLock disabled={!activeFocus}>
        {/**
         * the drawer will probably have buttons inside,
         * We're intentionally not adding role=button for now,
         * may so we're comfortable suppressing this rule
         */}
        {/* eslint-disable-next-line jsx-a11y/no-static-element-interactions */}
        <div
          aria-hidden={!isActive}
          className={componentClassName}
          onClick={handleOnClickOutside}
          onKeyDown={handleOnKeyDown}
          ref={ref}
          {...other}
        >
          <article
            aria-describedby={ariaDescribedBy}
            aria-labelledby={ariaLabelledBy}
            aria-modal={isActive}
            className={styles['drawer__window']}
            ref={windowRef}
            role="dialog"
            tabIndex={0} // eslint-disable-line jsx-a11y/no-noninteractive-tabindex
          >
            {header}
            {body}
            {footer}
          </article>
        </div>
      </FocusLock>
    </Portal>
  );
};

Drawer.Body = DrawerBody;
Drawer.Footer = DrawerFooter;
Drawer.Header = DrawerHeader;
