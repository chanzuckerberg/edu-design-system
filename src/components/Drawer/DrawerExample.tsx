import clsx from 'clsx';
import React, { ReactNode, useState, useRef, MutableRefObject } from 'react';
import { Drawer } from './Drawer';
import Button from '../Button';
import ButtonGroup from '../ButtonGroup';
import Heading from '../Heading';

export interface Props {
  /**
   * Child node(s) that can be nested inside component
   */
  children: ReactNode;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
}

/**
 * Primary UI component for user interaction
 */
export const DrawerExample = ({ className, ...other }: Props) => {
  const [drawerOpen, setDrawerOpen] = useState(false);
  const drawerButton = useRef() as MutableRefObject<HTMLButtonElement>;

  function openDrawerExample() {
    setDrawerOpen(true);
  }

  function closeDrawerExample(event: any) {
    setTimeout(() => {
      drawerButton.current.focus();
    }, 1);
    if (event) {
      event.preventDefault();
    }
    setDrawerOpen(false);
  }
  const componentClassName = clsx('tooltip-drawer', className, {});
  return (
    <div
      className={componentClassName}
      style={{ padding: '1rem', minHeight: '500px' }}
      {...other}
    >
      <Button onClick={openDrawerExample} ref={drawerButton} variant="primary">
        Open Drawer
      </Button>

      <Drawer
        ariaDescribedBy="drawer-description-1"
        ariaLabelledBy="drawer-heading-1"
        dismissible={true}
        isActive={drawerOpen}
        onClose={closeDrawerExample}
      >
        <Drawer.Header>
          <Heading id="drawer-heading-1" size="h2">
            Drawer title
          </Heading>
        </Drawer.Header>
        <Drawer.Body>
          <div className="fpo">Drawer body</div>
        </Drawer.Body>
        <Drawer.Footer>
          <ButtonGroup>
            <Button variant="primary">Button</Button>
            <Button onClick={closeDrawerExample}>Close Drawer</Button>
          </ButtonGroup>
        </Drawer.Footer>
      </Drawer>
    </div>
  );
};
