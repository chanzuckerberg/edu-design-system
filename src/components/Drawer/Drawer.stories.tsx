import { BADGE } from '@geometricpanda/storybook-addon-badges';
import { StoryObj, Meta } from '@storybook/react';
import React from 'react';

import { Drawer } from './Drawer';
import { DrawerExample } from './DrawerExample';
import { Button } from '../Button/Button';
import { ButtonGroup } from '../ButtonGroup/ButtonGroup';
import { Heading } from '../Heading/Heading';

export default {
  title: 'Organisms/Interactive/Drawer',
  component: Drawer,
  parameters: {
    badges: [BADGE.BETA],
  },
  args: {
    'aria-describedby': 'drawer-description-1',
    'aria-labelledby': 'drawer-heading-1',
    dismissible: true,
    isActive: true,
    children: (
      <>
        <Drawer.Header>
          <Heading id="drawer-heading-1" size="h2">
            Drawer title
          </Heading>
        </Drawer.Header>
        <Drawer.Body>
          <div className="fpo">Drawer body</div>
        </Drawer.Body>
        <Drawer.Footer>
          <ButtonGroup>
            <Button variant="primary">Button 1</Button>
            <Button>Button 2</Button>
          </ButtonGroup>
        </Drawer.Footer>
      </>
    ),
  },
  argTypes: {
    onClose: {
      action: 'on close',
    },
  },
} as Meta<Args>;

type Args = React.ComponentProps<typeof Drawer>;

/**
 * 1) Axe testing throws error due to Drawer component not being controlled, but this story is mostly for testing as the interactive story would only capture the toggle button.
 */
export const Default: StoryObj<Args> = {
  parameters: {
    axe: {
      disabledRules: ['aria-allowed-role'] /* 1 */,
    },
  },
};

/**
 * 1) No point snapping the button as this story is an interaction example.
 */
export const DefaultInteractive: StoryObj = {
  render: () => <DrawerExample />,
  parameters: {
    /* 1 */
    snapshot: { skip: true },
    chromatic: { disableSnapshot: true },
  },
};
