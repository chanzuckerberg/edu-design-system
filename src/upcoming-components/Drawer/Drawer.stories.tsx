import { BADGE } from '@geometricpanda/storybook-addon-badges';
import { Story, Meta } from '@storybook/react';
import React from 'react';

import { DrawerExample, Props } from './DrawerExample';

export default {
  title: 'Organisms/Interactive/Drawer',
  component: DrawerExample,
  parameters: {
    badges: [BADGE.BETA],
  },
} as Meta;

const Template: Story<Props> = (args) => <DrawerExample {...args} />;

export const Default = Template.bind({});
Default.args = {};
