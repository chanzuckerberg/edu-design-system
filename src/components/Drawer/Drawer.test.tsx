import { generateSnapshots } from '@chanzuckerberg/story-utils';
import { composeStories } from '@storybook/testing-react';
import { act, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import * as stories from './Drawer.stories';
const { DefaultInteractive } = composeStories(stories);

describe('<Drawer />', () => {
  generateSnapshots(stories, {
    getElement: async () => {
      const filters = await screen.findByRole('dialog');
      return filters.parentElement; // eslint-disable-line testing-library/no-node-access
    },
  });

  it('shows the dialog when the open dialog button is clicked', async () => {
    const user = userEvent.setup();
    render(<DefaultInteractive />);
    const openDrawerButton = await screen.findByRole('button', {
      name: 'Open Drawer',
    });
    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
    await act(async () => {
      await user.click(openDrawerButton);
    });
    expect(screen.getByRole('dialog')).toBeTruthy();
  });
});
