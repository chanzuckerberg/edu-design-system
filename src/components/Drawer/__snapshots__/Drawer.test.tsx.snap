// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer /> Default story renders snapshot 1`] = `null`;

exports[`<Drawer /> DefaultInteractive story renders snapshot 1`] = `
<div
  style="padding: 1rem; min-height: 500px;"
>
  <button
    class="clickable-style clickable-style--lg clickable-style--primary clickable-style--brand button button--primary"
    data-bootstrap-override="clickable-style-primary"
    type="button"
  >
    Open Drawer
  </button>
</div>
`;
